### PR TITLE
Makes death alarms now talk on medical coms

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
NO LONGER WILL YOUR PRIVEITY LAWS BE INGORED BY THE DEATH ALARM
PRASE BE!
Well inside the colony they broadcoast to medical
Well outside the colony the broadcoast to both blackshield and medical

### CODE REVIEW

**Comment by Kavrick** [comment]
This completely prevents prospectors, blackshield _and_ marshals from being able to utilize them, not sure why anyone would think this is a good idea honestly. Like what if someone dies outside the colony and blackshield needs to medevac them? I just dont see the point, death implants announcing over general comms has never caused any issue.

**Comment by nikothedude** [comment]
Well. It's against SOP I'm pretty sure.

**Comment by Yawet330** [comment]
-1 

shitpost PR made by trilby on a whim


Also because of what Kavrick said

We do not always have attentive and active medical staff, this will make it far harder to repeatedly track people when they die elsewhere on low-pop

I really don't know why this is being done.

**Comment by ThePainkiller** [comment]
**Solution:** Give corpsmen medical comms.

This makes it so that the **only** job on Blackshield that's supposed to handle Medevac has access to it when there's no medical staff. While I partly understand where Kavrick's coming from, why the fuck would PROSPECTORS need to know where someone died? Just stick together and don't wander too far away from the main group? You have Sawbones that just patch you up, not field medics that revive your peeps, that's Soteria's job.
Why would marshals need to know, besides the niche case of a protected witness getting murdered in colony grounds?
Why would ANY NON-MEDICAL JOB need to know the location of a CORPSE?

Otherwise, this goes in line with privacy laws and the policy of not stating coords of anyone without their consent on comms, it's not a shitpost PR. I think it's pretty valid, just not fully thought about on all fronts.

**Comment by cdb-is-not-good** [comment]
Not super sold on this for a lotta reasons. Some medical players DID want this but my main concern is that people will just stop using death alarms.

>Solution: Give corpsmen medical comms.

No.

**Comment by Spoopy11** [comment]
Yeah this plain doesn't make sense, when you're dead you care /very/ little for your privacy and more for your recovery.

Keeping  it to  medical coms will do the following:
Increase the possibility of this being missed by 1000x
     Medical SSD, or non-present  in low pop rounds
     IF AND IF, Corpsmen were given medical coms, or given the ability to hear: Corpsman not present in low pop rounds, or SSD
     Scavengers no longer hearing this will no longer attempt to recover them in the lack of the above

Overall?  Unnecessary, unthought out PR

**Comment by Leudoberct1** [comment]
Death alarms already respect privacy laws, as you *know* your status will be announced when you implant one, as well as being in a medical emergency, including death, already voiding your right to privacy in regards to your location. This also does kind of suck when medical are AFK or not around, so yeah.

**Comment by Churro55** [comment]
Everyone has presented very good arguments, I've nothing else to add other than I support them, and this PR shouldn't go through.

Read somewhere that this change was "supported by a lot of medical mains". Not only this is entirely false as I've yet to see a single med-main support it, but the only two people that have are Sci mains.

**Comment by Yawet330** [comment]
> ehhhhhhh
> 
> > Everyone has presented very good arguments, I've nothing else to add other than I support them, and this PR shouldn't go through.
> > Read somewhere that this change was "supported by a lot of medical mains". Not only this is entirely false as I've yet to see a single med-main support it, but the only two people that have are Sci mains (and I'm pretty sure one of them is just trolling).
> 
> Ayshe and Aeger both supported the change, and while i don't necessarily agree with it myself i do see the points behind it. The privacy argument is irrelevant, but what IS relevant is the fact that death alarms are simply too good - basically free, broadcast your last location from /anywhere/ to /everyone/ - which on the one hand yes, it's nice as it allows you to most readily be retrieved. On the other hand, it's yet another in a long line of things that utterly trivializes death and risk.

Then we should remove nanopaste preventing perma-death

This argument is shitty and will always be shitty, especially since this PR was submitted when several known and active bugs were confirmed to be occurring with mobs including firerate, target aggression, and otherwise balance issues with mobs which are STILL being smoothed out (such as spider venom going past blocks, same with egg injection).

I do not support this PR in any capacity and especially because it affects lowpop far more than it does high-pop, where the only real reason death alarms are used are during events to broadcast locations.

While I agree the 'free revive basically' aspect of them is pretty 'good', consider:

Cht'Mants
Psions
Any other race that can't take metal implants.

All of these already have Death Alarms not being applicable as a downside, and this would further reduce the severity of their nerfs (not like Cht'mant shrapnel toxin wasn't a joke anyway...)

I personally also don't really /care/ what two medical players have said. Death alarms have never quite affected them the same as it has other players. Sure, when they hop on prospector they've used them /sparingly/, but that isn't my point.

I respect Ayshe and Aeger, but their argument is based off of the fact that they've never really noticed the lowpop shifts (since they basically no longer play them, and this can be backed up by logs), where death-alarms have repeatedly allowed us to prevent people from just **flat out leaving the round**, like Painfeeler did earlier today, since their body wasn't retrieved for **two hours**, until someone finally logged on as SI doctor and checked sensors (as they didn't have a death alarm).


Look, the point and case is.

This will further make lowpop terrible, and assist in further killing our server pop at those times (usually EU), which makes the experience far less fun for everyone involved.

Privacy laws have NEVER been a concern with death alarms because AS PER LAW, someone in a critical condition is NOT CAPABLE OF MAKING CONCIOUS DECISION, and thus may be treated/located/have their coordinates said at SOTERIA's DISCRETION.

Taking a death alarm implant is YOUR conscious decision to void your right to locational privacy on death.

That is enough.

**Comment by cdb-is-not-good** [comment]
>Then we should remove nanopaste preventing perma-death

"If you nerf this you may as well remove revival altogether". No? That's a frankly ridiculous assertion to make. 

>This argument is shitty and will always be shitty, especially since this PR was submitted when several known and active bugs were confirmed to be occurring with mobs including firerate, target aggression, and otherwise balance issues with mobs which are STILL being smoothed out

As you said, PR was SUBMITTED, not accepted, and it was submitted partly at the suggestion of others. Again, this is not an argument against the PR itself. 

>I do not support this PR in any capacity and especially because it affects lowpop far more than it does high-pop, where the only real reason death alarms are used are during events to broadcast locations.

While it's important to consider the implication of PRs on lowpop, to an extent - we do not really largely balance AROUND lowpop. If there is absolutely no medical on and you're on a lowpop round, perhaps you should instead be careful or avoid danger?

>While I agree the 'free revive basically' aspect of them is pretty 'good', consider:
Cht'Mants
Psions
Any other race that can't take metal implants.

I.....okay? And? This is once again not actually an argument against the PR itself. The existence of psionics/Cht'Mants(the only two groups that can't have implants) doesn't....really invalidate the concerns about death alarms, the concerns about their efficacy, their range, etc. 

>All of these already have Death Alarms not being applicable as a downside

That's...a downside to them, not a downside to death alarms :V that's a downside you accept when you play a psion/chtmant, but it isn't a downside to death alarms, what? 

>I personally also don't really /care/ what two medical players have said. Death alarms have never quite affected them the same as it has other players. Sure, when they hop on prospector they've used them /sparingly/, but that isn't my point.

Is that....really an argument you want to make? By that token i could make the counter point that you ONLY care about these changes because they inconvenience you, a low-pop player(a timezone that, again - we do not balance around)

>This will further make lowpop terrible, and assist in further killing our server pop at those times (usually EU), which makes the experience far less fun for everyone involved.

Valid concern, offer an alternative. 

>Privacy laws have NEVER been a concern with death alarms

Correct, but irrelevant. I'm not arguing for this PR on those grounds. 

>Also secondary edit: CDB is entirely discounting the fact being revived gives both a temporary + perma stat down.

This might hold some weight if stats were relevant- In fact, YOU know as well as anyone that stats can be VERY easily gamed, and even without gaming them? These stat losses can be easily replaced.

**Comment by Yawet330** [comment]
> > Then we should remove nanopaste preventing perma-death
> 
> "If you nerf this you may as well remove revival altogether". No? That's a frankly ridiculous assertion to make.
> 
> > This argument is shitty and will always be shitty, especially since this PR was submitted when several known and active bugs were confirmed to be occurring with mobs including firerate, target aggression, and otherwise balance issues with mobs which are STILL being smoothed out
> 
> As you said, PR was SUBMITTED, not accepted, and it was submitted partly at the suggestion of others. Again, this is not an argument against the PR itself.
> 
> > I do not support this PR in any capacity and especially because it affects lowpop far more than it does high-pop, where the only real reason death alarms are used are during events to broadcast locations.
> 
> While it's important to consider the implication of PRs on lowpop, to an extent - we do not really largely balance AROUND lowpop. If there is absolutely no medical on and you're on a lowpop round, perhaps you should instead be careful or avoid danger?
> 
> > While I agree the 'free revive basically' aspect of them is pretty 'good', consider:
> > Cht'Mants
> > Psions
> > Any other race that can't take metal implants.
> 
> I.....okay? And? This is once again not actually an argument against the PR itself. The existence of psionics/Cht'Mants(the only two groups that can't have implants) doesn't....really invalidate the concerns about death alarms, the concerns about their efficacy, their range, etc.
> 
> > All of these already have Death Alarms not being applicable as a downside
> 
> That's...a downside to them, not a downside to death alarms :V that's a downside you accept when you play a psion/chtmant, but it isn't a downside to death alarms, what?
> 
> > I personally also don't really /care/ what two medical players have said. Death alarms have never quite affected them the same as it has other players. Sure, when they hop on prospector they've used them /sparingly/, but that isn't my point.
> 
> Is that....really an argument you want to make? By that token i could make the counter point that you ONLY care about these changes because they inconvenience you, a low-pop player(a timezone that, again - we do not balance around)
> 
> > This will further make lowpop terrible, and assist in further killing our server pop at those times (usually EU), which makes the experience far less fun for everyone involved.
> 
> Valid concern, offer an alternative.
> 
> > Privacy laws have NEVER been a concern with death alarms
> 
> Correct, but irrelevant. I'm not arguing for this PR on those grounds.
> 
> > Also secondary edit: CDB is entirely discounting the fact being revived gives both a temporary + perma stat down.
> 
> This might hold some weight if stats were relevant- In fact, YOU know as well as anyone that stats can be VERY easily gamed, and even without gaming them? These stat losses can be easily replaced.

The point wasn't that you should nerf revival altogether if this is nerfed, nay. The entire point was your line of thinking.

Death alarms by far usually do not make revival that easy, you still need to find players to actually go out and get them + retrieve their corpse, then you need to actually have someone with a high enough BIO stat (or god forbid the SI perk) to fix their mangled corpse and defib them, which - by the way, ATKs have a bio requirement of iirc like 140 and a cog requirement of 60 to use (in organ repair).

The reason I said I personally don't think Aeger/Ayshe's points apply is for the simple rule of SS13 HRP servers.

Lowpop. Kills. Servers.

You can witness it occasionally. People will NOT join Sojourn and instead sit in the lobby because there isn't enough people on for them to consider it fun. This only exacerbates the issue.

Again, this PR has basically zero grounds for it to stand upon, especially because the original reason was quoted 'Death alarms should respect privacy laws!'

> I.....okay? And? This is once again not actually an argument against the PR itself. The existence of psionics/Cht'Mants(the only two groups that can't have implants) doesn't....really invalidate the concerns about death alarms, the concerns about their efficacy, their range, etc.

Most implants suck long and hard through peoples Jorts, CDB. The only thing worthwhile has a tendency to be Death alarms, and their utility makes them a very /good/ implant to take if you're going exploring as a colonist/solo prospector, which again, due to the aforementioned issue of people not joining during lowpop (2 people ingame, 18 in lobby), yeahhh.

> Is that....really an argument you want to make? By that token i could make the counter point that you ONLY care about these changes because they inconvenience you, a low-pop player(a timezone that, again - we do not balance around)

As proven I barely play EU timezones and instead play on sat/sunday at 1-2 AM for highpop, I'm just giving a token-argument for the few times a week I do occasionally stoop by to try and stop the pop from tanking.

**Comment by MSieghardt** [comment]
I agree with this. Your death is a medical issue and it should only be transmitted to the Medical channel.

Merge and update.

**Review by SirRichardFrancis — COMMENTED** [review]
Probably could use same radio to speak in both channels.
Also only `radio_caller` is `qdel()`ed, `breaker_caller` is not for some reason.
Better set `null` spawn location and `qdel()` neither, as there is no point to queue reference-less object, it should be garbage collected anyway.
Also callers could be created before `switch()`, to cut duplicated code.

**Comment by cdb-is-not-good** [comment]
>[Yawet330]) commented [2 days ago]

Tl;Dr.

Going to be merging this on a probationary basis. If it sucks we'll tweak it or revert it, but we'll give this a chance given that if nothing else - several med mains and at least two staff devs have expressed support.

**Comment by Trilbyspaceclone** [comment]
Horrable pr -1
